sys/kern/src/arch/arm_m.rs

```diff
@@ -1210,7 +1210,10 @@ pub unsafe extern "C" fn DefaultHandler() {
                 .unwrap_or_else(|| panic!("unhandled IRQ {irq_num}"));
 
             let switch = with_task_table(|tasks| {
-                disable_irq(irq_num, false);
+                // This can only fail if the IRQ number is out of range, which
+                // in this case would mean the hardware is conspiring against
+                // us. So ignore it to ensure we don't generate a bogus check.
+                disable_irq(irq_num, false).ok();
```

Comment:
<s>I know you prefer this over `let _` because the latter is a bit of a footgun in other contexts, but there is a part of me that kind of wonders if 
```rust
fn_returning_result().ok();
```
optimizes worse than 
```rust
let _ = fn_returning_result();
```
because it's a function call, though hopefully LLVM can eliminate it...I might have to go check.</s>

**EDIT**: After spending some time in Compiler Explorer, I have been proven definitively wrong about this, in fact, the `opt-level=2` compiler is so smart about this that it correctly realizes that these are actually *the same function* and [doesn't even generate two separate functions for them](https://godbolt.org/z/hM75ThrYj):
```rust
#[no_mangle]
#[inline(never)]
pub extern fn ignore_err_ok() {
    returns_result().ok();
    // pretend the function then does other stuff
    core::hint::black_box(1);
}

#[no_mangle]
#[inline(never)]
pub extern fn ignore_err_underscore() {
    let _ = returns_result();
    // pretend the function then does other stuff
    core::hint::black_box(1);
}
```

So, I have seen the error of my ways and should not have doubted the use of `.ok()` to ignore `Result`s!.